### PR TITLE
feat(odata): InList filter, property paths, has_next/prev_page

### PR DESCRIPTION
**Pull request by ffedoroff**
## PR chain

This is **PR 1d of 8** in the resource-group feature train.

All 8 PRs together deliver the `resource-group` module. This PR extends the OData layer with capabilities the resource-group API needs: filtering by a list of values, filtering on nested hierarchy fields, and exposing pagination direction to API clients.

**Chain order:**
1a. #1401 docs/rg-1a-docs
1d. #1404 **→ feat/rg-1d-odata** *(this PR)*
1e. #1405 feat/rg-1e-db-security
2.  #1406 feat/rg-2-authz
3.  #1407 feat/rg-3-resource-group
4.  #1408 test/rg-4-tests

---

## Why this PR exists

The resource-group API exposes three OData-filterable collections: types, groups, and memberships. Implementing those endpoints revealed three gaps in the existing OData stack:

**1. `IN (...)` filter operator was missing.**
Group queries need to filter by a set of type codes: `$filter=type_code in ('department','team')`. Without `FilterNode::InList`, the API would have to accept only equality filters and force clients to make N requests for N values.

**2. Property paths with `/` were not parsed.**
Group hierarchy queries expose fields like `hierarchy/depth` and `hierarchy/breadth` (depth and width within the closure table). The OData 4.01 spec allows `/` as a navigation path separator (`firstMemberExpr`), but the parser only handled flat identifiers. Without this fix, `$filter=hierarchy/depth le 3` would fail to parse.

**3. Clients had no way to know if more pages exist.**
`PageInfo` returned `next_cursor` and `prev_cursor` but did not expose boolean `has_next_page` / `has_previous_page` flags. Frontend and SDK clients had to check cursor nullability themselves, which is error-prone. The flags make pagination intent explicit with no extra DB query (derived from cursor presence).

---

## Summary

- Add `FilterOp::In` and `FilterNode::InList` to the OData filter tree
- Support OData 4.01 property paths with `/` separator in the parser
- Add `has_next_page` / `has_previous_page` to `PageInfo`
- Update SeaORM filter compiler and cursor pagination to match

---

## Changes

**`libs/modkit-odata/src/filter.rs`**
- `FilterOp::In` — new operator variant
- `FilterNode::InList { field, values }` — built from `field in (v1, v2, ...)` expressions; field must be an identifier (not a computed expression), values must be literals. Both invariants are enforced at parse time with clear error messages.

**`libs/modkit-odata/src/odata_parse.rs`**
- `property_path` rule replaces `identifier` in value expressions — allows `/`-separated segments as required by OData 4.01 ABNF §5.1.1.15
- `identifier` rule preserved for contexts where paths are not valid (function names, etc.)

**`libs/modkit-odata/src/odata_parse_tests.rs`**
- New tests: `odata_navigation_path_identifier`, `odata_navigation_path_with_and` — cover single-segment and compound `AND` expressions with paths

**`libs/modkit-odata/src/page.rs`**
- `PageInfo.has_next_page: bool` — `true` when `next_cursor` is `Some`
- `PageInfo.has_previous_page: bool` — `true` when `prev_cursor` is `Some`

**`libs/modkit-odata/src/problem_mapping.rs`** — error mapping extended for `FilterOp::In` context messages

**`libs/modkit-db/src/odata/sea_orm_filter.rs`**
- `FilterNode::InList` → `Expr::col(column).is_in(sea_values)` with per-value type coercion via `odata_value_to_sea_value`
- `has_next_page` / `has_previous_page` populated from cursor presence after query

**`libs/modkit-db/src/odata/core.rs`** — same pagination flag population for the alternate query path

**`libs/modkit-sdk/src/pager.rs`** — `Pager` exposes the new flags to SDK consumers

**`libs/modkit/tests/odata_select.rs`** — extended select tests

---

## Test plan

- [ ] `cargo test -p modkit-odata` — new parser tests pass
- [ ] `cargo test -p modkit-db` — `InList` generates correct `WHERE col IN (...)` SQL
- [ ] `$filter=name in ('a','b')` compiles to `WHERE name IN ('a', 'b')`
- [ ] `$filter=hierarchy/depth ge 0` parses without error
- [ ] `has_next_page` is `true` only when a next cursor exists
- [ ] Existing OData filter tests pass unchanged

## Summary by CodeRabbit

* **New Features**
  * Added support for the OData IN operator (list membership) with validation against empty lists.
  * Extended filter parsing to accept slash-separated navigation/property paths.

* **Bug Fixes**
  * More specific cursor validation error messages for distinct failure types.

* **Tests**
  * Added parser tests for navigation-path identifiers, comparisons, logical conjunctions, and invalid path edge cases.

* **Documentation**
  * Reordered OpenAPI OData extension blocks for list endpoints.

### CODE REVIEW

File: libs/modkit-db/src/odata/sea_orm_filter.rs [hunk 1]
```diff
@@ -581,12 +589,17 @@ where
 
     let items = rows.into_iter().map(model_to_domain).collect();
 
+    let has_next_page = next_cursor.is_some();
```
[the remark this answers is not in the document]
Comment (ffedoroff) [reply libs/modkit-db/src/odata/sea_orm_filter.rs]: Reverted — same as core.rs, restored to plain struct construction.

**Comment by ffedoroff** [comment]
## Latest fixes (all review findings addressed)

**Nitpick: Empty IN-list guard** (`sea_orm_filter.rs:157`)
Fixed — added `values.is_empty()` check before building the condition. Returns a clear error `"IN list must not be empty"` instead of producing invalid SQL `col IN ()`.

**Nitpick: Ambiguous last-segment match** (`filter.rs:from_name`)
Fixed — fallback resolution now collects all matches and fires `debug_assert!` if more than one field shares the same terminal segment. Added doc comment explaining the limitation.

**Inline: IN-list field-kind coercion** (`sea_orm_filter.rs:159-162`)
Already fixed in previous commit — `odata_value_to_sea_value_for_kind()` uses `field.kind()` for correct i64/f64/Decimal coercion.

**Inline: Property path resolution** (`odata_parse.rs:82`, `filter.rs:325`)
Already fixed in previous commit — `from_name()` tries full path first, falls back to last segment.

**Nitpick: Malformed navigation path tests** (`odata_parse_tests.rs`)
Fixed — added 3 new tests: `leading_slash_rejected`, `empty_segment_rejected`, `double_slash_rejected`. All pass.

**Comment by ffedoroff** [comment]
Fixed: ambiguous property path resolution now returns None (→ UnknownField) when multiple fields share the same terminal segment, instead of silently picking the first match. No debug_assert — safe in both debug and release builds.

**Review by MikeFalcon77 — APPROVED** [review]

**Review by MikeFalcon77 — APPROVED** [review]
